From 6a7aff11e70b61754e1e7ad1983ef9fb6471412f Mon Sep 17 00:00:00 2001
Message-Id: <6a7aff11e70b61754e1e7ad1983ef9fb6471412f.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Mon, 7 Jul 2014 13:42:32 +0100
Subject: [PATCH 135/312] FOR_UPSTREAM [VPG]: drm/i915: Add
 WaProgramL3SqcReg1Default:bdw

For performance, program the default initial value of L3SqcReg1 on BDW
to 0x784000:

L3SQ High Priority Credit Initialization = 2 (00001b).
L3SQ General Priority Credit Initialization = 30 (01111b).

Issue: VIZ-4086
Change-Id: I8a12968fa7f3ed54fe11e272188f5074e4b2b17c
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    4 ++++
 drivers/gpu/drm/i915/intel_pm.c |    3 +++
 2 files changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 606279b..6c09364 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4739,6 +4739,10 @@ enum punit_power_well {
 #define  HDC_FENCE_DESTINATION_TO_SLM_DISABLE	(1<<14)
 #define  HDC_FORCE_NON_COHERENT			(1<<4)
 
+/* WaProgramL3SqcReg1Default */
+#define GEN8_L3SQCREG1				0xb100
+#define  GEN8_L3SQCREG1_DEFAULT_VALUE	0x784000
+
 /* WaCatErrorRejectionIssue */
 #define GEN7_SQ_CHICKEN_MBCUNIT_CONFIG		0x9030
 #define  GEN7_SQ_CHICKEN_MBCUNIT_SQINTMOB	(1<<11)
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index b74aaed..8a61278 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5605,6 +5605,9 @@ static void gen8_init_clock_gating(struct drm_device *dev)
 	/* Wa4x4STCOptimizationDisable:bdw */
 	I915_WRITE(CACHE_MODE_1,
 		   _MASKED_BIT_ENABLE(GEN8_4x4_STC_OPTIMIZATION_DISABLE));
+
+	/* WaProgramL3SqcReg1Default:bdw */
+	I915_WRITE(GEN8_L3SQCREG1, GEN8_L3SQCREG1_DEFAULT_VALUE);
 }
 
 static void haswell_init_clock_gating(struct drm_device *dev)
-- 
1.7.9.5

